File: Leetle/tests/reference_solutions/javascript/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
function twoSum(nums, target) {
    const numMap = new Map();
    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];
        if (numMap.has(complement)) {
            return [numMap.get(complement), i];
        }
        numMap.set(nums[i], i);
    }
    return [];
}

// Main function to handle input from stdin
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').trim();
const lines = input.split('\n');

// Parse array (first line)
const nums = JSON.parse(lines[0].trim());

// Parse target (second line)
const target = parseInt(lines[1].trim());

const result = twoSum(nums, target);
console.log(JSON.stringify(result));

File: Leetle/tests/reference_solutions/python/two_sum.py
def two_sum(nums, target):
    """
    Given an array of integers nums and an integer target,
    return indices of the two numbers such that they add up to target.
    """
    num_map = {}
    for i, num in enumerate(nums):
        complement = target - num
        if complement in num_map:
            return [num_map[complement], i]
        num_map[num] = i
    return []

# Main function to handle input from stdin
if __name__ == "__main__":
    import sys
    input_data = sys.stdin.read().strip()
    lines = input_data.split('\n')

    # Parse array (first line)
    nums_str = lines[0].strip()
    import ast
    nums = ast.literal_eval(nums_str)

    # Parse target (second line)
    target = int(lines[1].strip())

    result = two_sum(nums, target)
    print(str(result).replace(' ', ''))
